File: RobWork/example/cpp/ex-load-workcell.cpp
#include <rw/loaders/WorkCellLoader.hpp>
#include <rw/models/WorkCell.hpp>

using rw::loaders::WorkCellLoader;
using rw::models::WorkCell;

int main(int argc, char** argv)
{
    if (argc != 2) {
        std::cout << "Usage: " << argv[0] << " <workcell>" << std::endl;
        return 1;
    }

    const std::string file = argv[1];
    WorkCell::Ptr workcell = WorkCellLoader::Factory::load(file);
    if (workcell.isNull()) {
        std::cout << "WorkCell could not be loaded." << std::endl;
        return 1;
    }

    std::cout << "Workcell " << workcell->getName();
    std::cout << " successfully loaded." << std::endl;
    return 0;
}

File: RobWork/example/lua/ex-load-workcell.lua
require("sdurw_core")
require("sdurw_models")
require("sdurw")
using("sdurw")

if #arg ~= 1 then
    print("Usage: lua ex-load-workcell.lua <workcell>")
    return 1
end

workcell = WorkCellLoaderFactory.load(arg[1])
if workcell:isNull() then
    print("WorkCell could not be loaded")
    return 1
end

print("Workcell " .. workcell:getName() .. " successfully loaded.");